### PR TITLE
Fixes #31712 - Extend API params for generating registration command

### PR DESCRIPTION
**Pull request by stejskalleos**
For: https://github.com/theforeman/foreman/pull/8271

### CODE REVIEW

**Comment by chris1984** [comment]
Starting to test

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Review by chris1984 — COMMENTED** [review]
@stejskalleos I am getting this error when trying to start up the dev server:

```ruby
3727] ! Unable to load application: NameError: uninitialized constant RegistrationCommandController
```

**Comment by stejskalleos** [comment]
@chris1984 What branch do you have on Foreman? This PR is extension of https://github.com/theforeman/foreman/pull/8271, without it doesn't work.

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
@stejskalleos oh I was on develop. ok I will test with that.

**Review by chris1984 — APPROVED** [review]
ACK, tested with UI and was able to register a host and also with curl:

```bash
curl -X POST https://devel.toledo.satellite.lab.eng.rdu2.redhat.com/api/registration_command --user admin:changeme -H 'Content-Type: application/json'

[vagrant@devel ~]$ curl -X POST https://devel.toledo.satellite.lab.eng.rdu2.redhat.com/api/registration_command --user admin:changeme -H 'Content-Type: application/json'
{"registration_command":"curl  -X GET 'https://devel.toledo.satellite.lab.eng.rdu2.redhat.com/register' -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjo0LCJpYXQiOjE2MTIxOTI5ODQsImp0aSI6Ijg4MDM4NDA1NzFmZTU5MWY0ZjIzMTRjZWVjOWZiN2I0OWM3ODc5OTNkNzQxY2VjMTVlYWJjNjcxM2VlMWE2NGQiLCJleHAiOjE2MTIyMDczODQsInNjb3BlIjoicmVnaXN0cmF0aW9uI2dsb2JhbCByZWdpc3RyYXRpb24jaG9zdCJ9.ZhWpe516naVMjgUlVxPhqUsCKhJyvHpIA57WH-NCI_c' | bash"}
```

Will merge once the Foreman PR is merged otherwise tests will fail

File: app/controllers/katello/concerns/api/v2/registration_command_controller_extensions.rb
Comment (ezr-ondrej) [inline app/controllers/katello/concerns/api/v2/registration_command_controller_extensions.rb:5]:
```suggestion
        module RegistrationCommandsControllerExtensions
```

File: lib/katello/engine.rb
Comment (ezr-ondrej) [inline lib/katello/engine.rb:179]:
```suggestion
      ::RegistrationCommandsController.include Katello::Concerns::RegistrationCommandsControllerExtensions
```

File: lib/katello/engine.rb
Comment (ezr-ondrej) [inline lib/katello/engine.rb:220]:
```suggestion
      ::Api::V2::RegistrationCommandsController.include Katello::Concerns::Api::V2::RegistrationCommandsControllerExtensions
```

File: lib/katello/plugin.rb
Comment (ezr-ondrej) [inline lib/katello/plugin.rb:287]:
```suggestion
  extend_page 'registration_commands/_form' do |cx|
```

File: lib/katello/plugin.rb
Comment (ezr-ondrej) [inline lib/katello/plugin.rb:288]:
```suggestion
    cx.add_pagelet :global_registration, name: N_('Katello'), partial: 'foreman/hosts/registration_commands', priority: 100, id: 'katello'
```

File: test/controllers/foreman/registration_command_controller_test.rb
Comment (ezr-ondrej) [inline test/controllers/foreman/registration_command_controller_test.rb:3]:
```suggestion
class RegistrationCommandsControllerTest < ActionController::TestCase
```

**Review by ezr-ondrej — COMMENTED** [review]
This is now ublocked, just needs renaming after the core PR renamed the Controller.

**Review by ezr-ondrej — APPROVED** [review]
I believe this is ready.

**Comment by stejskalleos** [comment]
@chris1984 rebased & updated

**Comment by chris1984** [comment]
Starting to test again and will merge if it looks good

**Review by chris1984 — APPROVED** [review]
ACK.

Works well:

```shell
[root@dhcp182-243 ~]# curl --insecure -X GET 'https://devel.toledo.satellite.lab.eng.rdu2.redhat.com/register?activation_key=food&location_id=2&organization_id=1' -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjo0LCJpYXQiOjE2MTM1ODA4NTQsImp0aSI6ImZiOTU3YzJiYzljN2E4ODlkZmZjMjkyMDEzMTdmZWUzODFiMmZhYjFlZTY1MDFiY2U1NTA2N2JkNTIwODkwMjIiLCJleHAiOjE2MTM1OTUyNTQsInNjb3BlIjoicmVnaXN0cmF0aW9uI2dsb2JhbCByZWdpc3RyYXRpb24jaG9zdCJ9.n4SetcHTEGHbgV-TyRynCgOAXl7MOkxYXJaftoHVUcQ' | bash
```
```shell
Complete!
The system has been registered with ID: d4e2c292-51a3-4846-8d0a-aa228735f495
The registered system name is: dhcp182-243.gsslab.rdu2.redhat.com
Installed Product Current Status:
Product Name: Red Hat Enterprise Linux Server
Status:       Subscribed
```

**Pull request merged**
